packages/wonder-blocks-accordion/src/components/accordion.tsx

```diff
@@ -162,56 +161,63 @@ const Accordion = React.forwardRef(function Accordion(
         newSectionsOpened[index] = newOpenedValueAtIndex;
         setSectionsOpened(newSectionsOpened);
 
-        // Keep track of the currently focused section for keyboard navigation.
-        setCurrentlyFocusedSection(index);
-
         if (childOnToggle) {
             childOnToggle(newOpenedValueAtIndex);
         }
     };
 
-    const handleSectionFocus = (index: number) => {
-        setCurrentlyFocusedSection(index);
-    };
-
-    /** Keyboard navigation for keys: ArrowUp, ArrowDown, Home, and End.
-     *
-     * Note that we don't have to use `setCurrentlyFocusedSection` in this
-     * function because the focus is handled by the browser + the
-     * section's onFocus handler (the `handleSectionFocus` function above).
+    /**
+     * Keyboard navigation for keys: ArrowUp, ArrowDown, Home, and End.
      */
     const handleKeyDown = (event: React.KeyboardEvent) => {
+        // From https://www.w3.org/WAI/ARIA/apg/patterns/accordion/
+        // - Down Arrow: If focus is on an accordion header, moves focus to the next accordion header. If focus is on the last accordion header, either does nothing or moves focus to the first accordion header.
+        // - Up Arrow: If focus is on an accordion header, moves focus to the previous accordion header. If focus is on the first accordion header, either does nothing or moves focus to the last accordion header.
+        // - Home: When focus is on an accordion header, moves focus to the first accordion header.
+        // - End: When focus is on an accordion header, moves focus to the last accordion header.
+
+        const currentlyFocusedHeaderIndex = childRefs.findIndex(
+            (ref) => ref.current === document.activeElement,
+        );
+
+        // If the currently focused element is not a header, do nothing.
+        if (currentlyFocusedHeaderIndex === -1) {
+            return;
+        }
+
```

Comment:
This is awesome! Do you think you could add unit test(s) for the new logic in `accordion.test.tsx` under the "Keyboard navigation" section? 

Specifically, I think we'd want to add a test for making sure that nothing happens if the focus isn't on a header. We could also check that keyboard nav works for stuff inside the content (like the dropdowns in the story you added) within an accordion.

Reply:
Tests added!